src/mscorlib/shared/System/IO/Path.cs

```diff
@@ -118,6 +137,26 @@ public static string GetExtension(string path)
             return string.Empty;
         }
 
+        public static ReadOnlySpan<char> GetExtension(ReadOnlySpan<char> path)
```

Comment:
don't duplicate the body of `GetExtension(string)`. Duplication leads to maintenance pain. Instead make the string one call to the span one using `path.AsReadOnlySpan`